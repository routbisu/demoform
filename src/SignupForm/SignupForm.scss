.trial-form {
  margin: auto;
  max-width: 800px;
  padding: 10px;
  border: 1px solid #eaeaea;

  h2 {
    size: 12px;
    color: darkcyan;
    text-align: center;
  }

  .form-group {
    margin: 10px 0px;
  }

  input, textarea {
    border: 1px solid #e1e1e1;
    display: block;
    padding: 10px;
    width: 100%;
    box-sizing: border-box;

    &:disabled {
      background: #eaeaea;
      cursor: not-allowed;
    }
  }

  .label {
    font-size: 11px;
    color: darkcyan;
    width: 70px;
    display: inline-block;
    margin-bottom: 5px;
  }

  .message {
    vertical-align: top;
  }

  .btn-submit {
    border: none;
    background: darkcyan;
    color: #ffffff;
    padding: 10px 25px;
    width: 100%;
    outline: none;
    cursor: pointer;
  }

  .btn-loading {
    border: 1px solid darkcyan;
    background: #f9f9f9;
  }

  .success-message {
    margin-top: 10px;
    border: 1px dotted #cff3d7;
    background: #f9fafb;
    padding: 10px 15px;
    font-size: 11px;
    color: #72777a;
    ul {
      margin-top: 10px;
      margin-left: 17px;
      line-height: 1.7em;
      li {
        span {
          font-weight: bold;
        }
      }
    }
  }

  @media only screen and (min-width: 600px) {
    input, textarea {
      display: inline-block;
      width: 400px;
    }
  }
}