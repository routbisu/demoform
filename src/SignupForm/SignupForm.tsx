import React, { Component } from 'react';
import './SignupForm.scss';
import Loader from '../images/loader.svg';
import classNames from 'classnames';
import { any } from 'prop-types';
import SuccessMessage from './SuccessMessage/SuccessMessage';
import FormGroup from './FormGroup/FormGroup';

interface state {
  isLoading: boolean;
  formSubmitted: boolean;
  name: string;
  email: string;
  message: string;
}

interface props {
  signUpComplete: any;
}

export class SignupForm extends Component<props, state> {
  constructor(props: any) {
    super(props);
    this.state = {
      isLoading: false,
      formSubmitted: false,
      name: '',
      email: '',
      message: ''
    };
  }

  signup = (evt: any) => {
    evt.preventDefault();
    this.props.signUpComplete();

    // Validate fields
    if (!(this.state.name && this.state.email && this.state.message)) {
      alert('Please fill all the required fields');
    } else {
      // Make an AJAX call to sign up
      this.setState({ isLoading: true, formSubmitted: false });
      setTimeout(() => {
        this.setState({ isLoading: false, formSubmitted: true });
      }, 3000);
    }
  };

  onChangeHandler = (evt: any) => {
    const fieldName: any = evt.target.name;
    this.setState({
      [fieldName]: evt.target.value
    } as any);
  };

  render() {
    const btnClasses = classNames('btn-submit', { 'btn-loading': this.state.isLoading });

    return (
      <div className="trial-form">
        <h2>Trial Form</h2>

        {this.state.formSubmitted ? <SuccessMessage {...this.state} /> : null}

        <form onSubmit={this.signup} id="frmTrial">
          <FormGroup onChangeHandler={this.onChangeHandler} label="Name" name="name" isLoading={this.state.isLoading} />
          <FormGroup
            onChangeHandler={this.onChangeHandler}
            label="Email"
            name="email"
            type="email"
            isLoading={this.state.isLoading}
          />
          <FormGroup
            onChangeHandler={this.onChangeHandler}
            label="Message"
            name="message"
            type="textarea"
            isLoading={this.state.isLoading}
          />
          <button type="submit" className={btnClasses}>
            {this.state.isLoading ? <img src={Loader} alt="Loading" /> : 'Sign Up'}
          </button>
        </form>
      </div>
    );
  }
}

export default SignupForm;
