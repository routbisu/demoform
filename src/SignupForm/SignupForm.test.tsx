import React, { Component } from 'react';
import { mount } from 'enzyme';
import SignupForm from './SignupForm';

describe('Signup Form', () => {
  let mockFn = jest.fn();

  const evt = {
    preventDefault: () => {}
  };

  const form = mount(<SignupForm signUpComplete={mockFn} />);

  it('renders properly', () => {
    expect(form).toMatchSnapshot();
  });

  it('Check signup', () => {
    form.setState({
      name: 'Biswa',
      email: 'bis@gmail.com',
      message: 'Some message'
    });
    form.find('#frmTrial').simulate('submit', evt);
    expect(mockFn).toBeCalled();
  });
});
