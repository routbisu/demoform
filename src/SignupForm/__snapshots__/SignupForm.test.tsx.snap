// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form renders properly 1`] = `ReactWrapper {}`;
